packages/netlify-cms-core/src/components/Collection/Sidebar.js

```diff
@@ -133,7 +133,12 @@ class Sidebar extends React.Component {
             value={query}
           />
         </SearchContainer>
-        <SidebarNavList>{collections.toList().map(this.renderLink)}</SidebarNavList>
+        <SidebarNavList>
+          {collections
+            .toList()
+            .filter(collection => collection.hidden === false)
```

Comment:
This condition will never be true as. `collection` is an immutablejs instance so you need to access its fields via `collection.get('hidden') !== true`

Reply:
Understood. Will do.